.circleci/config.yml

```
@@ -0,0 +1,126 @@
+              chmod +x $HOME/yq/yq &&\
+              echo 'export PATH=$HOME/yq/yq:$PATH' >> $BASH_ENV
+  install_gh:
+    description: "Install GitHub Cli"
+    steps:
+      - run:
+          name: Install gh
+          command: |
+            curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg
+            echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null
+            sudo apt update
+            sudo apt install gh
+  set_sdk_version:
+    description: "Set Daml SDK version environment version"
+    steps:
+      - run:
+          name: Set Daml SDK version
+          command: echo 'export SDK_VERSION=$(yq e '.sdk-version' daml.yaml)' >> $BASH_ENV
+  run_release:
+    description: "Run Release"
+    steps:
+      - run:
+          name: "Get target release version"
+          command: echo 'export VERSION=$(yq e '.version' daml.yaml)' >> $BASH_ENV
+      - run:
+          name: "Validate target release version"
+          command: |
+            if [ `git tag` == $VERSION ] ; then
+              echo "New tag detected ${VERSION}. Releasing..."
+            else
+              echo "Tag ${VERSION} already exists; Skipping Release."
+              circleci-agent step halt
+            fi
+      - run:
+          name: "Tag and push to GitHub"
+          command: gh release create v${VERSION} .daml/dist/daml-ctl-${VERSION}.dar --generate-notes
+      - store_artifacts:
+          path: .daml/dist/daml-ctl-${VERSION}.dar
+          destination: daml-ctl-dar-${VERSION}
+
+jobs:
+  build:
+    executor: daml-docker-executor
+    steps:
+      - checkout
+      - install_yq
+      - restore_cache:
+          keys:
+            - daml-ctl-{{ checksum "daml.yaml" }}
+      - set_sdk_version
+      - install_sdk
+      - run:
+          name: Daml build
+          command: daml build
+      - run:
+          name: Daml test
+          command: daml test
+      - save_cache:
+          paths:
+            - .daml
+          key: daml-ctl-{{ checksum "daml.yaml" }}
+  release:
+    executor: daml-docker-executor
+    steps:
+      - checkout
+      - install_yq
+      - install_gh
+      - restore_cache:
+          keys:
+            - daml-ctl-{{ checksum "daml.yaml" }}
+      - run_release
+
+workflows:
+  version: 2
+  daml_ctl_build:
+    jobs:
+      - build:
+          filters:
+            branches:
+              ignore:
+                - master
+  daml_ctl_release:
+    jobs:
+      - build:
+          filters:
+            branches:
+              only:
+                - master
+      - release:
+          context: github-fin-eng-context
+          requires:
+            - build
+          filters:
+            branches:
+              only:
+                - master
```

```suggestion
  daml_ctl_build:
    jobs:
      - build

  daml_ctl_release:
    jobs:
      - release:
          context: github-fin-eng-context
          requires:
            - build
          filters:
            branches:
              only:
                - master
```
That's what I'm suggesting :point_up: (not sure syntax is correct) . Does that make sense?

---

Have you tested that ? What you changed is that the `build` job now runs in its own workflow and that the `release` is now depending on a job thats ran **outside** of its workflow. I suspect that won't work but I haven't checked the docs or tested it locally.

---

No haven't tested 

> (not sure syntax is correct)

---

Doesn't seem to be the case - `We don’t have a way to declare dependencies between workflows in a pipeline.` (https://discuss.circleci.com/t/run-another-workflow-after-a-workflow-succeeds/36294). Also https://discuss.circleci.com/t/reference-a-job-between-workflows/40700

---

It's up to you - I just thought it was weird. Maybe that's the way everyone does it.